sdk/src/wallet/wallet/operations/address_generation.rs

```diff
@@ -109,13 +112,173 @@ impl Wallet {
+                    options
+                });
+                // Generate without prompt to be able to display it
+                let address = self
+                    .secret_manager
+                    .read()
+                    .await
+                    .generate_ed25519_addresses(
+                        self.coin_type.load(Ordering::Relaxed),
+                        account_index,
+                        address_index..address_index + 1,
+                        changed_options,
+                    )
+                    .await?;
+
+                let bech32_hrp = self.get_bech32_hrp().await?;
+
+                self.emit(
+                    account_index,
+                    WalletEvent::LedgerAddressGeneration(AddressData {
+                        address: address[0].to_bech32(bech32_hrp),
+                    }),
+                )
+                .await;
+            }
+
+            // Generate with prompt so the user can verify
+            self.secret_manager
+                .read()
+                .await
+                .generate_ed25519_addresses(
+                    self.coin_type.load(Ordering::Relaxed),
+                    account_index,
+                    address_index..address_index + 1,
+                    options,
+                )
+                .await?
+        } else {
+            self.secret_manager
+                .read()
+                .await
+                .generate_ed25519_addresses(
+                    self.coin_type.load(Ordering::Relaxed),
+                    account_index,
+                    address_index..address_index + 1,
+                    options,
+                )
+                .await?
+        };
+
+        Ok(*address
+            .first()
+            .ok_or(crate::wallet::Error::MissingParameter("address"))?)
+    }
+}
+
+#[cfg(feature = "stronghold")]
+impl Wallet<crate::client::secret::stronghold::StrongholdSecretManager> {
+    /// Generate an address without storing it
+    /// ```ignore
+    /// let public_addresses = wallet
+    ///     .generate_ed25519_address(
+    ///         0,
+    ///         false,
+    ///         0,
+    ///         None,
+    ///     )
+    ///     .await?;
+    /// ```
+    pub async fn generate_ed25519_address(
+        &self,
+        account_index: u32,
+        address_index: u32,
+        options: Option<GenerateAddressOptions>,
+    ) -> crate::wallet::Result<Ed25519Address> {
+        let address = self
+            .secret_manager
+            .read()
+            .await
+            .generate_ed25519_addresses(
+                self.coin_type.load(Ordering::Relaxed),
+                account_index,
+                address_index..address_index + 1,
+                options,
+            )
+            .await?;
+
+        Ok(*address
+            .first()
+            .ok_or(crate::wallet::Error::MissingParameter("address"))?)
     }
+}
+
+impl Wallet<MnemonicSecretManager> {
```

Comment:
Is there actually a good reason why anyone should use the Wallet with a SecretManager not via the enum, if there is a variant for it? Because to me it looks like an option nobody will use but just adds a lot more code

Reply:
Well, I'd rather have some duplicated code than completely remove the ability to use the wallet like this.

Reply:
Why "remove" the ability? It was never there and you can still use a MnemonicSecretManager in the enum without this

Reply:
Done